pkg/agents/team6/behaviours.go

```diff
@@ -4,46 +4,52 @@ import (
 	"math"
 )
 
+// Updates agent social motive:
+// First updates behaviour weights, then computes behaviour change based on weights and input parameters (HP, floor)
 func (a *CustomAgent6) updateBehaviour() {
+	a.updateBehaviourWeights()
 	aConf := a.config
 	behaviourMax, behaviourMin := a.behaviourRange()
 
-	hpScore := 1 - float64(a.HP())/100.0
+	// Utility func for HP
+	hpScore := 1 - float64(a.HP())/float64(a.HealthInfo().MaxHP)
 	floor := a.Floor() + 1
 
+	// Update agent's guess of how many floors there are in the tower
 	if floor > a.maxFloorGuess {
 		a.maxFloorGuess = floor + 1
 	}
 
+	// Utility func for floor
 	fdNum := float64(floor) / float64(a.maxFloorGuess) * aConf.lambda
```

Comment:
Just checking that `float64(a.maxFloorGuess)` is never 0?

Reply:
Nice spot - we always initialise as our current floor and then update if we get a score bigger - not possible to get 0

Reply:
Cool